### ISSUE TITLE
Validator ignores custom format

### ISSUE THREAD

**Issue by meikandamoorthi** [everit-org/json-schema#30]
When validating against this JSON schema
{
	"$schema":"http://json-schema.org/draft-04/schema#",
	"type":"object",
	"properties": {
		"id": {
			"type": "string",
			"maxLength": 10,
        	        "minLength": 3,
			"format": "evenlength"
		}
	},
	"required": ["id"]
}
JSON Document is:
{
	"id": "everitorg"
}

Try with your documentation [example](https://github.com/everit-org/json-schema#format-validators). But i got no response/error. Only string validation is working fine(success/error). But custom format "evenlength" is not working. why?

**Comment by erosb** [comment]
Hello,
could you please also post your java code?

**Comment by meikandamoorthi** [comment]
**The custom FormatValidator will look something like this:**

```
import java.util.Optional;
import org.everit.json.schema.FormatValidator;
public class EvenCharNumValidator implements FormatValidator {
	@Override
	  public Optional<String> validate(final String subject) {
	    if (subject.length() % 2 == 0) {
	    	System.out.println(subject);
	      return Optional.empty();
	    } else {
	    	System.out.println(subject);
	      return Optional.of(String.format("the length of srtring [%s] is odd", subject));
	    }
	  }
}
```

**The schema loader configuration:**

```
        File schemaFile = new File("/Users/UserName/Downloads/schema.json");
		BufferedReader schemaBR = new BufferedReader(new FileReader(schemaFile));
		StringBuffer schemaStrbuf = new StringBuffer();
		String schemaCurrentLine;
		while ((schemaCurrentLine = schemaBR.readLine()) != null)
		{
			schemaStrbuf.append(schemaCurrentLine);
		}
		
		File dataFile = new File("/Users/UserName/Downloads/data.json");
		BufferedReader dataBR = new BufferedReader(new FileReader(dataFile));
		StringBuffer dataStrbuf = new StringBuffer();
		String dataCurrentLine;
		while ((dataCurrentLine = dataBR.readLine()) != null)
		{
			dataStrbuf.append(dataCurrentLine);
		}

		JSONObject jsonSchema = new JSONObject(new JSONTokener(schemaStrbuf.toString()));

		SchemaLoader schemaLoader = SchemaLoader.builder()
				.schemaJson(jsonSchema) // rawSchema is the JSON representation of the schema utilizing the "evenlength" non-standard format
				.addFormatValidator("evenlength", new EvenCharNumValidator()) // the EvenCharNumValidator gets bound to the "evenlength" keyword
				.build();
		
		Schema schema = schemaLoader.load().build(); // the schema is created using the above created configuration
		System.out.println(schema);


		JSONObject jsonSubject = new JSONObject(new JSONTokener(dataStrbuf.toString()));

		try
		{
			schema.validate(jsonSubject);  // the document validation happens here
		}
		catch(ValidationException ve)
		{
			System.out.println(ve.getLocalizedMessage());
		}
```

**Comment by meikandamoorthi** [comment]
@erosb any update?

**Comment by erosb** [comment]
Hello @meikandamoorthi , I've fixed it yesterday in 9830d2946b28e4c7cc3f77278af8e793a3c68f1f .
